### PR TITLE
Frame updates

### LINKED ISSUE #82: Door: allow negative frame depth
1. Many times, door/window frame is deeper than the wall. Currently we don't support that. We only support frame to extrude (with +ive frame depth) but never intrude.
2. Also, I currently door depth is independent of frame. I think it would be better to move door (according to door depth) relative to the frame.

### PR DESCRIPTION
**Pull request by luckykadam**
1. allow negative frame depth
2. limit frame thickness to strictly >0

fixes #82 , #58

### CODE REVIEW

**Comment by ranjian0** [comment]
Reviewing!

File: core/door/door_props.py [hunk 1]
```diff
@@ -8,17 +8,17 @@
...
         description="Thickness of door Frame",
     )
 
     frame_depth: FloatProperty(
         name="Frame Depth",
-        min=0.0,
+        min=-1.0,
         max=1.0,
-        default=0.0,
+        default=0.05,
```
Comment (ranjian0) [inline core/door/door_props.py:21]: Setting `0.05` as default feels un-natural. The doors and windows now look extremely busy. I think `0.0` is better here, and its nicely midway between min and max

Comment (ranjian0) [reply core/door/door_props.py:21]: This default only looks appropriate on multigroup but not doors and windows.

Comment (luckykadam) [reply core/door/door_props.py]: reset default frame depth to 0.0

File: core/door/door_types.py [hunk 2]
```diff
@@ -82,18 +82,18 @@ def create_door_frame(bm, face, prop):
 
     # add depths
     if prop.add_arch:
-        # frame depth
-        all_faces = [door_face] + [arch_face] + frame_faces
-        all_faces, surrounding_faces = extrude_face_region(bm, all_faces, -prop.frame_depth, normal)
-        door_face, arch_face, frame_faces = all_faces[0], all_faces[1], all_faces[2:] + surrounding_faces
+        if prop.frame_depth != 0.0:  # frame depth
+            all_faces = [door_face] + [arch_face] + frame_faces
```
Comment (ranjian0) [inline core/door/door_types.py:86]:
```python
[door_face, arch_face] + frame_faces
```
is better here. Only one list created

File: core/window/window_types.py [hunk 3]
```diff
@@ -84,18 +84,18 @@ def create_window_frame(bm, face, prop):
 
     # add depths
     if prop.add_arch:
-        # frame depth
-        all_faces = [window_face] + [arch_face] + frame_faces
-        all_faces, surrounding_faces = extrude_face_region(bm, all_faces, -prop.frame_depth, normal)
-        window_face, arch_face, frame_faces = all_faces[0], all_faces[1], all_faces[2:] + surrounding_faces
+        if prop.frame_depth != 0.0:  # frame depth
+            all_faces = [window_face] + [arch_face] + frame_faces
```
Comment (ranjian0) [inline core/window/window_types.py:88]: same here

**Comment by ranjian0** [comment]
Noticed that blocks starting with 
```python
if prop.frame_depth != 0.0:
```
all look quite similar in `doot_types.py`, `window_types.py` and `multigroup_types.py`

Perhaps they can be refactored in a function and placed in `core/frame.py` which currently only has one function. Should also reduce the size of the `create_*_frame` functions in the corresponding modules which seem to be getting uncomfortably long.

**Comment by ranjian0** [comment]
Also, the lonely function in `core/frame.py` seems to be useless now.

**Pull request merged**